keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -9,105 +9,201 @@
+<<[/UNRESOLVED]>>
 
 ### Non-Goals
 
-- Reading TLS serving certificates and key from a socket or reloading of the API server with new cert and key
+- Reading TLS serving certificates and key from a socket or reloading of kube-apiserver with new cert and key
 - Reading any other certificates from a file
 
 ## Proposal
 
 ### Preserve existing behavior
 
-The API server flags `--service-account-key-file` and `--service-account-signing-key-file` will continue be used for reading from files.
+The kube-apiserver flags `--service-account-key-file` and `--service-account-signing-key-file` will continue to be used for reading from files.
 
-### Updates to API server token generation
+### New API
 
-As of Kubernetes v1.13.2, the API server uses the functions `JWTTokenGenerator` and `JWTTokenAuthenticator`. New types that implement the `TokenGenerator` interface and support token validation will be added to `k8s.io/kubernetes/pkg/serviceaccount/`.
+I'm proposing creating a new versioned grpc API (ExternalJWTSigner) under `k8s.io/kubernetes/pkg/serviceaccount`. This will be similar to how the KMS envelope encryption has an API at `k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v1beta1/service.proto`. It will have methods to sign JWTs and fetch supported keys (proto in [Design Details](#design-details) section). 
 
-### New API
+### Plugins
+
+- A signer plugin will be a client to the new grpc API thus providing support for external signing.
+- A public key cache will integrate with the same grpc to fetch supported public keys.
+
+### Updates for token generation
+
+As of Kubernetes `v1.13.2`, The Kubernetes Control Plane config accepts `ServiceAccountIssuer` which is an interface of type `serviceaccount.TokenGenerator` as defined in `pkg/serviceaccount/jwt.go`. This enables defining alternate implementations of token generators. Thus, the default TokenGenerator can be replaced by one that integrates with ExternalJWTSigner.
+
+### Updates for supported public keys
+
+Starting ` v1.31` (via. the [PR](https://github.com/kubernetes/kubernetes/pull/125177) by liggit@), `ServiceAccountPublicKeys` in kube-apiserver config was replaced by `ServiceAccountPublicKeysGetter` interface. The static keys held by `ServiceAccountPublicKeys` were hence moved to be managed by the implementation called `StaticPublicKeysGetter`. This presents an opportunity to add a new implementation, one that will get public keys from ExternalJWTSigner depending on how the cluster is configured.
+
+In the same PR, `OpenIDMetadataServer` was updated to accept implementation of `OpenIDMetadataProvider` as defined in  `pkg/serviceaccount/openidmetadata.go`. `OpenIDMetadataProvider` interacts with `ServiceAccountPublicKeysGetter` for serving supported keys which enables flexible replacing `StaticPublicKeysGetter` with a dynamic key getter (will integrate with ExternalJWTSigner) depending on how cluster is configured.
```

Comment:
can simplify to drop history and PR links, just link to the current interface(s), describe the current static file implementations in a sentence, and indicate we'd have an alternate implementation backed by the grpc client and key cache

Reply:
done